docs/configuration/uds-operator.md

```diff
@@ -207,6 +207,48 @@ spec:
 
 This configuration does not create a secret in the cluster and instead tells the UDS Operator to create a public client (one that requires no auth secret) that enables the `oauth2.device.authorization.grant.enabled` flow and disables the standard redirect auth flow.  Because this creates a public client configuration that deviates from this is limited - if your application requires both the Device Authorization Grant and the standard flow this is currently not supported without creating two separate clients.
 
+### Creating a UDS Package with a Service Account Roles client
+
+Some applications may need to access resources / obtain OAuth tokens on behalf of *themselves* vice users. This may be needed to allow API access to Authservice protected applications (outside of a web browser). This is commonly used in machine-to-machine authentication for automated processes. This type of grant in OAuth 2.0 is known as the [Client Credentials Grant](https://oauth.net/2/grant-types/client-credentials/) and is supported in a UDS Package with the following configuration:
+
+```yaml
+apiVersion: uds.dev/v1alpha1
+kind: Package
+metadata:
+  name: client-cred
+  namespace: argo
+spec:
+  sso:
+    - name: httpbin-api-client
+      clientId: httpbin-api-client
+      standardFlowEnabled: false
+      serviceAccountsEnabled: true
+
+      # By default, Keycloak will not set the audience `aud` claim for service account access token JWTs.
+      # You can optionally add a protocolMapper to set the audience.
+      # If you map the audience to the same client used for authservice, you can enable access to authservice protected apps with a service account JWT.
+      protocolMappers:
+        - name: audience
+          protocol: "openid-connect"
+          protocolMapper: "oidc-audience-mapper"
+          config:
+            included.client.audience: "uds-core-httpbin" # Set this to match the app's authservice client id
+            access.token.claim: "true"
+            introspection.token.claim: "true"
+            id.token.claim: "false"
+            lightweight.claim: "false"
+            userinfo.token.claim: "false"
+
+      # Optionally, store the client secret a Kubernetes secret
```

Comment:
We have made it required that this client will be `publicClient: false` (has a secret), but this comment refers to `secretName` and `secretTemplate`. As far as I can tell `secretName` is optional but will default to `secretName` if unset.

Reply:
It would make sense to me if it wasn't optional.  One use case where it might not matter as much is if the client is outside the cluster.  Then you can find a means to pass them the client secret and either grab it from a kubernetes secret or as the keycloak admin.

Reply:
Yeah I guess to clarify my comment on the comment 😄: The kubernetes secret will always be created unless the client is a public client. So regardless of whether secret name and template are specified here, a k8s secret will be created - it will just end up with a default name and format if you don't provide these (ref code [here](https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/controllers/keycloak/client-sync.ts#L186-L205)). Not to get hung up on one line here in a docs comment but "store the client secret as a Kubernetes secret" is not really optional/conditional on setting these fields was my main point. Comment might be better stated as "optionally set the name and template for the generated kubernetes secret"?